tests/run-fail/collatz.c

```diff
@@ -0,0 +1,36 @@
+
+#include <limits.h>
+#include <stdbool.h>
+#include <stdint.h>
+
+#include "decaf.h"
+
+uint32_t __attribute__((noinline)) modulo(uint32_t x, uint32_t y) {
+  return x % y;
+}
+
+uint32_t __attribute__((noinline)) udiv(uint32_t x, uint32_t y) {
+  return x / y;
+}
+
+uint32_t __attribute__((noinline)) collatz(uint32_t x) {
+  uint32_t cnt = 0;
+  while (x > 1) {
+    cnt += 1;
+
+    uint32_t k1 = udiv(x, 2);
+    uint32_t k2 = 3 * x + 1;
+
+    x = (modulo(x, 2) == 0) ? k1 : k2;
+    decaf_assume(cnt < 10);
+  }
+
+  return cnt;
+}
+
+void test(uint32_t n, uint32_t expected) {
+  uint32_t nc = collatz(n);
+
+  if (nc == expected)
```

Comment:
Why is `nc` being compared to `expected` here?

Shouldn't `nc` always be `<= 2`?

Reply:
Maybe worthwhile to add just a small comment to explain what's happening there

Reply:
This was so I could see the count that it was producing when I was debugging it. I'll remove it since it doesn't matter to the test case itself.

Reply:
Also, this produces output if `nc > 2` which is the goal of the test case.